playlists/views.py

```
@@ -37,22 +39,38 @@ class AddSpotifyPlaylist(generics.CreateAPIView):
+class UpdateSpotifyPlaylist(generics.UpdateAPIView):
+    serializer_class = AddSpotifyPlaylistSerializer
+    permission_classes = [permissions.IsAuthenticated]
+
+    def get_object(self):
+        season_id = self.request.data.get('seasonId')
+        user = self.request.user
+        try:
+            playlist = Playlist.objects.get(season_id=season_id, owner=user)
+            return playlist
+        except Playlist.DoesNotExist:
+            raise NotFound("No existing playlist found for this season.")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)